operation/manager.go

```diff
@@ -132,7 +149,6 @@ func (sm *SingleOperationManager) cancelInLock(ctx context.Context) {
 	}
 
 	op.cancelFunc()
-	<-op.waitCh
```

Comment:
just remove waitCh entirely from the op struct

Reply:
done